rollout/controller.go

```diff
@@ -882,7 +882,7 @@ func (c *rolloutContext) getReferencedAnalysisTemplatesFromRef(templateRefs *[]v
 	templates := make([]*v1alpha1.AnalysisTemplate, 0)
 	clusterTemplates := make([]*v1alpha1.ClusterAnalysisTemplate, 0)
 	for _, templateRef := range *templateRefs {
-		if templateRef.ClusterScope {
+		if templateRef.ClusterScope != nil && *templateRef.ClusterScope {
```

Comment:
How do you feel about adding methods on the type to do `templateRef.IsClusterScope()` ?